src/browser/js/main.js

```diff
@@ -19,7 +19,145 @@ import $ from 'jquery';
  * 4. Добавить возможность отфильтровать тудушки по названию
  *   При вводе в строку поиска нужно оставить видимыми только тудушки, в которых содержится введенная подстрока
  */
+const getToDoTemplate = (params) => {
+    return `<li
+                    class="list-group-item d-flex justify-content-between align-items-center"  data-list-todo-id="${params.id}"
+                >
+                    <span>${params.title}</span>
+                    <i class="far fa-trash-alt delete" data-todo-id="${params.id}"></i>
+                </li>`
+}
 
-const initPageLogic = () => {};
+const createToDo = (params) => {
+    const lastToDo = document.querySelector('.list-group-item:last-of-type');
+    const listToDo = document.getElementById('list-group-todos');
+
+    if (lastToDo != null)
+    {
+        lastToDo.insertAdjacentHTML('afterend', getToDoTemplate(params))
+    }
+    else
+    {
+        listToDo.insertAdjacentHTML('afterbegin', getToDoTemplate(params));
+    }
+
+}
+
+
+const getToDoFetch = async () => {
+    const response = await fetch('https://jsonplaceholder.typicode.com/todos?_limit=10');
+    const toDoRowData = await response.json();
+
+
+    return toDoRowData.map((rawBackendElement) => ({
+        title: rawBackendElement.title,
+        id: rawBackendElement.id,
+        completed: rawBackendElement.completed,
+    }))
+}
+
+const addNewToDo = async (params) => {
+    const response = await fetch('https://jsonplaceholder.typicode.com/todos',
+                {
+                    method: 'POST',
+                    body: JSON.stringify({title: params.title})
+                });
+    if (response.ok)
+    {
+       createToDo({ title: params.title });
+    }
+    /*const result = await response.json();
+    console.log(result.message);
+    alert(result.message);*/
+}
+
+const deleteToDo = (element) => {
+    if (element.matches('[data-todo-id]')) {
+        if (element.closest('[data-list-todo-id]')) {
+            element.closest('[data-list-todo-id]').remove();
+        }
+    }
+}
+
+const filterToDolist = (item) => {
+    const allToDo = document.querySelectorAll('[data-list-todo-id]');
+    allToDo?.forEach((element) => {
+        for (const child of element.children)
+        {
+            if (child.matches('span'))
+            {
+                if (String(child.innerHTML).indexOf(item.value) >= 0)
+                {
+                  //console.log(element);
+                  element.classList.remove('filtred');
+                }
+                else
+                {          
+                    console.log(element);
+                    //element.remove();
+                   //element.style
+                   //console.log(child.innerHTML);
+                  element.classList.add('filtred');
+                };
+            }
+        }
+        
```

Comment:
Почему лучше не пользоваться свойствами .children .childnodes и т. д. - потому что верстка структурно может изменится. 

Например: добавят ещё одну обертку из div, добавят ещё один спан где будут лейблы которыми мы помечаем определенные группы дел

Даже если нерабочий функционал сразу заметят, это все равно потраченное время на поддержку JS решения.

Более стабильным вариантом на мой взгляд будет:
1. Добавление data атрибута для элемента где непосредственно содержится текст тудушки
2. Селект этого элемента от корня карточки
3. Чтение значения из него и уже логика по скрытию

```suggestion
        if (element.querySelector('[data-todo-name]').textContent.includes(item.value)) {
            element.classList.remove('filtred');
        } else {
            element.classList.add('filtred');
        }
```